chapter/1/rpc-reviewed-dont-keep.md

```diff
@@ -0,0 +1,619 @@
+passing links to a global shared memory location is not impossible but rather
+dependent on the type of system (see *Applications* section for detail).
+
+Originally, RPC was developed as a synchronous, language-specific marshalling
+service with a custom network protocol to outsource computation {% cite
+implementingrpc --file rpc %}. It had registry-system to register all the
+servers. One of the earliest RPC-based system {% cite implementingrpc --file rpc %}
+was implemented in the Cedar programming language in early 1980's. The goal of
+this system was to provide similar programming semantics as local procedure
+calls. Developed for a LAN network with an inefficient network protocol and a
+*serialization* scheme to transfer information using the said network protocol,
+this system aimed at executing a *procedure*(also referred as *method* or a
+*function*) in a remote *address space*. The single-thread synchronous client
+and the server were written in an old *Cedar* programming language with a
+registry system used by the servers to *bind*(or register) their procedures. The
+clients used this registry system to find a specific server to execute their
+*remote* procedures.
+
+Modern RPC-based systems are language-agnostic, asynchronous, load-balanced
+systems. Authentication and authorization to these systems have been added as
+needed along with other security features. Most of these systems have
+fault-handling built into them as modules.
+
+RPC programs have a network (or a communication channel), therefore, they need
+to handle remote errors and be able to communication information successfully.
+Error handling generally varies and is categorized as *remote-host* or *network*
+failure handling. Depending on the type of the system, and the error, the caller
+(or the callee) return an error and these errors can be handled accordingly. For
+asynchronous RPC calls, it's possible to specify events to ensure progress.
+
+RPC implementations use a *serialization*(also referred to as *marshalling* or
+*pickling*) scheme on top of an underlying communication protocol (traditionally
+TCP over IP). These *serialization* schemes allow both the caller *caller* and
+*callee* to become language agnostic allowing both these systems to be developed
+in parallel without any language restrictions. Some examples of serialization
+schemes are JSON, XML, or Protocol Buffers {% cite grpc --file rpc %}.
+
+RPC allows different components of a larger system to be developed independently
+of one another. The language-agnostic nature combined with a decoupling of some
+parts of the system allows the two components (caller and callee) to scale
+separately and add new functionalities. This independent scaling of the system
+might lead to a mesh of interconnected RPC *services* facilitating one another.
+
+### Examples of RPC
+
+RPC has become very predominant in modern systems. In the simplest RPC systems,
```

Comment:
Here's the link that starts the video at that specific point, mentioning that internally the order is around 10^10 RPC calls/sec: 

   https://www.youtube.com/watch?v=xb8u2s7cxzg&t=486s